msticpy/nbtools/timeline.py

```diff
@@ -0,0 +1,940 @@
+    range_tool : bool, optional
+        Show the the range slider tool (default is True)
+    height : int, optional
+        The height of the plot figure
+        (the default is auto-calculated height)
+    width : int, optional
+        The width of the plot figure (the default is 900)
+    color : str
+        Default series color (default is "navy"). This is overridden by
+        automatic color assignments if plotting a grouped chart
+    kind : Union[str, List[str]]
+        one or more glyph types to plot., optional
+        Supported types are "circle", "line" and "vbar" (default is "vbar")
+
+    Returns
+    -------
+    figure
+        The bokeh plot figure.
+
+    """
+    reset_output()
+    output_notebook()
+
+    height: int = kwargs.pop("height", None)
+    width: int = kwargs.pop("width", 900)
+    ref_time: Any = kwargs.pop("ref_time", None)
+    title: str = kwargs.pop("title", None)
+    time_column = kwargs.get("x", time_column)
+    group_by: str = kwargs.get("group_by", None)
+    sort_by: str = kwargs.pop("sort_by", None)
+    show_yaxis: bool = kwargs.pop("yaxis", True)
+    show_range: bool = kwargs.pop("range_tool", True)
+    color: str = kwargs.get("color", "navy")
+    legend_column: str = kwargs.pop("legend_column", None)
+    legend_pos: str = kwargs.pop("legend", None)
+    kind: Any = kwargs.pop("kind", ["vbar"])
+    plot_kinds = kind if isinstance(kind, list) else [kind]
+
+    if not ref_time:
+        ref_event = kwargs.get("alert", kwargs.get("ref_event", None))
+        if ref_event:
+            ref_time = getattr(ref_event, "StartTimeUtc", None)
+            if not ref_time:
+                ref_time = getattr(ref_event, "TimeGenerated", None)
+
+    graph_df, group_count_df, tool_tip_columns, series_count = _create_data_grouping(
+        data, source_columns, time_column, group_by, sort_by, color
+    )
+
+    # build the tool tips from columns (excluding these)
+    tool_tip_items = [(f"{col}", f"@{col}") for col in tool_tip_columns]
+    hover = HoverTool(tooltips=tool_tip_items, formatters={"Tooltip": "printf"})
+
+    # Create the Plot figure
+    title = title if title else "Timeline"
+    min_time = graph_df[time_column].min()
+    max_time = graph_df[time_column].max()
+    start_range = min_time - ((max_time - min_time) * 0.1)
+    end_range = max_time + ((max_time - min_time) * 0.1)
+    height = height if height else _calc_auto_plot_height(series_count)
+
+    plot = figure(
+        x_range=(start_range, end_range),
+        min_border_left=50,
+        plot_height=height,
+        plot_width=width,
+        x_axis_label="Event Time",
+        x_axis_type="datetime",
+        x_minor_ticks=10,
+        y_axis_label=y,
+        tools=[hover, "xwheel_zoom", "box_zoom", "reset", "save", "xpan"],
+        toolbar_location="above",
+        title=title,
+    )
+    plot.yaxis.visible = show_yaxis
+    plot.ygrid.minor_grid_line_color = "navy"
+    plot.ygrid.minor_grid_line_alpha = 0.3
```

Comment:
These minor lines don't really add anything except clutter the graph, suggest we set minor_grid_line_color = None